### PR TITLE
[#7285] Live Server: disallow team name to be valid email

### LINKED ISSUE #7285: Student: submit a team response: the student's team members should also be added to the session's responding students list
Currently only students who submit a response on behalf of the team are added to the list. This could create a situation where team members have to resubmit the same response in order to be considered a respondent.

### PR DESCRIPTION
**Pull request by junming403**
Part of #7285

Disallow team name to be valid emails.

### CODE REVIEW

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 1]
```diff
@@ -144,6 +144,8 @@
             + "The value of '${fieldName}' field should be no longer than ${maxLength} characters.";
     public static final String INVALID_NAME_ERROR_MESSAGE =
             ERROR_INFO + " " + HINT_FOR_CORRECT_FORMAT_FOR_INVALID_NAME;
+    public static final String TEAM_NAME_IS_VALID_EMAIL_ERROR_MESSAGE =
+            "The field " + TEAM_NAME_FIELD_NAME + " is not acceptable to TEAMMATES as it appears to be a valid email";
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/util/FieldValidator.java]: "As it appears to be a valid email" does not tell the user immediately why it cannot be a team name.

**Comment by wkurniawan07** [comment]
Closing and re-opening since PR was targeted at wrong branch previously.

File: src/main/java/teammates/common/util/FieldValidator.java (hunk 1 above)
Comment (junming403) [reply src/main/java/teammates/common/util/FieldValidator.java]: How about "is not acceptable to TEAMMATES as team name can't be a valid email"?

**Comment by junming403** [comment]
Ready for review

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 2]
```diff
@@ -144,6 +144,8 @@
             + "The value of '${fieldName}' field should be no longer than ${maxLength} characters.";
     public static final String INVALID_NAME_ERROR_MESSAGE =
             ERROR_INFO + " " + HINT_FOR_CORRECT_FORMAT_FOR_INVALID_NAME;
+    public static final String TEAM_NAME_IS_VALID_EMAIL_ERROR_MESSAGE =
+            "The field " + TEAM_NAME_FIELD_NAME + " is not acceptable to TEAMMATES as team name can't be a valid email";
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/util/FieldValidator.java]: "As team name can't be a valid email" sounds quite weird. Suggest the following rephrase:

```txt
... is not acceptable to TEAMMATES as the suggested value for team name can be mis-interpreted as an email.
```

Also, `team name` is the value for the constant `TEAM_NAME_FIELD_NAME` so let's reuse it.

**Comment by junming403** [comment]
Updated, ready for review!

**Comment by damithc** [comment]
Received this new error from the live server today. Could it be related to this PR?
>java.lang.NullPointerException
teammates.common.util.Logger severe: Unexpected exception caught by ControllerServlet : java.lang.NullPointerException at teammates.common.datatransfer.questions.FeedbackContributionQuestionDetails.getTeamSubmissionArray(FeedbackContributionQuestionDetails.java:517) at teammates.common.datatransfer.questions.FeedbackContributionQuestionDetails.getQuestionResultsStatisticsHtmlQuestionView(FeedbackContributionQuestionDetails.java:253) at teammates.common.datatransfer.questions.FeedbackContributionQuest...

**Comment by junming403** [comment]
Yes Prof, I think it might be due to teamName = `null`.  I will create a PR to fix this issue shortly. Sorry for the trouble.

**Comment by junming403** [comment]
Hi prof, I think it's not relevant.

**Pull request merged**